controllers/kyma_controller.go

```diff
@@ -0,0 +1,98 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/eventing-auth-manager/api/v1alpha1"
+	kymav1beta1 "github.com/kyma-project/lifecycle-manager/api/v1beta1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
+)
+
+// KymaReconciler reconciles a Kyma resource
+type KymaReconciler struct {
+	client.Client
+	Scheme               *runtime.Scheme
+	errorRequeuePeriod   time.Duration
+	defaultRequeuePeriod time.Duration
+}
+
+func NewKymaReconciler(c client.Client, s *runtime.Scheme, errorRequeuePeriod time.Duration, defaultRequeuePeriod time.Duration) *KymaReconciler {
+	return &KymaReconciler{
+		Client:               c,
+		Scheme:               s,
+		errorRequeuePeriod:   errorRequeuePeriod,
+		defaultRequeuePeriod: defaultRequeuePeriod,
+	}
+}
+
+// +kubebuilder:rbac:groups=operator.kyma-project.io,resources=kymas,verbs=get;list;watch
+// +kubebuilder:rbac:groups=operator.kyma-project.io,resources=eventingauths,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=operator.kyma-project.io,resources=eventingauths/status,verbs=get;list
+func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling Kyma resource")
+
+	kyma := &kymav1beta1.Kyma{}
+	err := r.Client.Get(ctx, req.NamespacedName, kyma)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			return ctrl.Result{
+				RequeueAfter: r.defaultRequeuePeriod,
```

Comment:
I think that in this case we should not set a RequeueAfter, otherwise this object will be constantly reconciled, even after deletion.
I think I'll remove the `defaultRequeuePeriod` in my integration test PR, since this was only introduced to control the reconcile for the tests. And it's a better option set the [SyncPeriod](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager@v0.14.6#Options.SyncPeriod) oder the manager used for the tests and keep the default behaviour.

```suggestion
```